### PR TITLE
Extract role map method

### PR DESCRIPTION
**Pull request by s-e-v**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -83,11 +83,24 @@ class ANDGatewayClient {
...
   getUserRoleMap() {
-    return true;
+    const token = this._getTokenCookie();
+    if (token == undefined) {
+      throw new Error("Authorisation token not found");
+    } else {
+      const payload = decodeJWT(token);
+      const appsAccess = payload["http://sso.and.digital/roles"];
+      if (appsAccess == undefined) {
+        return undefined;
```
Comment (oskanberg) [inline src/index.js]: Perhaps return `[]`?

Comment (s-e-v) [reply src/index.js]: Good idea, they'll only see an empty screen basically but not kicked out, different story for company which I think should stay undefined.

File: src/index.js [hunk 2]
```diff
@@ -82,6 +82,27 @@ class ANDGatewayClient {
     }
   }
 
+  /**
+   * Obtains the user's role map (Apps access) from the authorisation token, to be used when the user's company is not AND Digital.
+   * @returns {array} - Returns an array with the name of the apps the user has access to.
+   * @returns {undefined} - Returns undefined if the roles key is not found in the authorisation token.
```
Comment (oskanberg) [inline src/index.js]: Not anymore

File: src/index.js [hunk 3]
```diff
@@ -64,22 +64,43 @@ class ANDGatewayClient {
...
-      const company = payload["http://sso.and.digital/company"];
-      if (company == undefined) {
-        return undefined;
-      } else {
-        return company;
-      }
     }
+    const payload = decodeJWT(token);
+    const company = payload["http://sso.and.digital/company"];
+    // Error
```
Comment (oskanberg) [inline src/index.js]: I think you can remove these `error` comments

**Pull request merged**